.github/workflows/xlsx-to-csv.yml

```
@@ -10,50 +9,73 @@ on:
+            full_path="$repo/$xlsx_file"
+            if [ -f "$full_path" ]; then
+              csv_file="$repo/mosip-data/mosip_master_csv/csv/$(basename "${xlsx_file%.xlsx}.csv")"
+              xlsx2csv "$full_path" > "$csv_file"
+              echo "Converted $xlsx_file -> $csv_file"
+            else
+              echo "WARNING: $full_path not found, skipping."
+            fi
+          done <<< "$changed_files"
+
+      - name: Commit and push only if CSVs actually changed
+        run: |
```

actionlint (1.7.12)

[error] "github.event.pull_request.head.ref" is potentially untrusted. avoid using it directly in inline scripts. instead, pass it through an environment variable. see https://docs.github.com/en/actions/reference/security/secure-use#good-practices-for-mitigating-script-injection-attacks for more details

(expression)